internal/checker/terraform/rules/azure/azurerm_iothub_no_public_network_access.rego

```diff
@@ -0,0 +1,62 @@
+# METADATA
+# title: "Verify that public network access is disabled for Azure IoT Hub"
+# description: "Validate that Azure IoT Hub restricts public network access, enhancing security by preventing unauthorized external connections"
+# scope: package
+# related_resources:
+# - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/iothub
+# custom:
+#   id: CB_TFAZR_106
+#   severity: MEDIUM
+package lib.terraform.CB_TFAZR_106
+
+import future.keywords.in
+
+supportedResources := ["azurerm_iothub"]
+
+isvalid(block) {
+	block.Type == "resource"
+	some label in block.Labels
+	label in supportedResources
+}
+
+resource[resource] {
+	some block in pass
+	resource := concat(".", block.Labels)
+}
+
+resource[resource] {
+	some block in fail
+	resource := concat(".", block.Labels)
+}
+
+pass[resource] {
```

Comment:
is line 32 to line 36 really needed?

Reply:
Thank you for noticing this unnecessary code. This has been removed.